adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationContext.cs

```diff
@@ -261,10 +261,24 @@ public async Task<AuthenticationResult> AcquireTokenByDeviceCodeAsync(DeviceCode
             IPlatformParameters parameters,
             UserIdentifier userId, string extraQueryParameters, string claims)
         {
-            return await this.AcquireTokenWithClaimsCommonAsync(resource, new ClientKey(clientId), redirectUri,
-                    parameters,
-                    userId, extraQueryParameters, this.CreateWebAuthenticationDialog((PlatformParameters)parameters), claims)
```

Comment:
The fix is moving this.CreateWebAuthenticationDialog((PlatformParameters)parameters) to the handler classes, because this class does not have a requestContext object.